ml-proto/test/memory.wast

```diff
@@ -31,6 +32,10 @@
   (module (memory 1 2 (segment 0 "a") (segment 2 "b") (segment 1 "c")))
   "data segment not disjoint and ordered"
 )
+(assert_invalid
+  (module (memory 0 65536))
```

Comment:
How about also adding a test with a value that overflows the 32-bit (resp 64-bit range), to check that there is no accidental wrap-around?